### PR TITLE
I ran ./lint_clean_files.sh and fixed the errors﻿

### PR DESCRIPTION
**Pull request by matchuek**
## Motivation and Context
https://github.com/Bash-it/bash-it/issues/1696
## How Has This Been Tested?
I don't see necessity for testing these changes.
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] If I have added a new file, I also added it to ``clean_files.txt`` and formatted it using ``lint_clean_files.sh``.
- [ ] I have added tests to cover my changes, and all the new and existing tests pass.

### CODE REVIEW

**Comment by NoahGorny** [comment]
Hi @matchuek, I am delighted that you opened up a PR in an effort to clean up Bash-it :smile: 

I see that the main difference is replacing `which` with `command -v`. I did not see any `shellcheck` warnings on this one, can you send here the error that you received when running `lint_clean_files.sh`?

**Comment by matchuek** [comment]
Hello @NoahGorny,

here is the output of `./lint_clean_files.sh`:

    Trim Trailing Whitespace....................................................Passed
    Fix End of Files............................................................Passed
    Check for merge conflicts...................................................Passed
    Mixed line ending...........................................................Passed
    Check for added large files.................................................Passed
    Check for conflict markers and core.whitespace errors.......................Passed
    Test shell scripts with shellcheck..........................................Failed
    - hook id: shellcheck
    - exit code: 1
    
    In themes/brainy/brainy.theme.bash line 139:
    		|| [ -z "$(which todo.sh)" ] && return
                               ^---^ SC2230: which is non-standard. Use builtin 'command -v' instead.
    
    For more information:
      https://www.shellcheck.net/wiki/SC2230 -- which is non-standard. Use builti...
    
    In themes/base.theme.bash line 88:
    GIT_EXE=$(which git 2> /dev/null || true)
              ^---^ SC2230: which is non-standard. Use builtin 'command -v' instead.
    
    
    In themes/base.theme.bash line 89:
    P4_EXE=$(which p4 2> /dev/null || true)
             ^---^ SC2230: which is non-standard. Use builtin 'command -v' instead.
    
    
    In themes/base.theme.bash line 90:
    HG_EXE=$(which hg 2> /dev/null || true)
             ^---^ SC2230: which is non-standard. Use builtin 'command -v' instead.
    
    
    In themes/base.theme.bash line 91:
    SVN_EXE=$(which svn 2> /dev/null || true)
              ^---^ SC2230: which is non-standard. Use builtin 'command -v' instead.
    
    
    In themes/base.theme.bash line 403:
    	if which rvm &> /dev/null; then
               ^---^ SC2230: which is non-standard. Use builtin 'command -v' instead.
    
    
    In themes/base.theme.bash line 412:
    	if which rbenv &> /dev/null; then
               ^---^ SC2230: which is non-standard. Use builtin 'command -v' instead.
    
    For more information:
      https://www.shellcheck.net/wiki/SC2230 -- which is non-standard. Use builti...
    
    In completion/available/consul.completion.bash line 6:
    	complete -C "$(which consul)" consul
                           ^---^ SC2230: which is non-standard. Use builtin 'command -v' instead.
    
    For more information:
      https://www.shellcheck.net/wiki/SC2230 -- which is non-standard. Use builti...
    
    In themes/atomic/atomic.theme.bash line 151:
    		|| [ -z "$(which todo.sh)" ] && return
                               ^---^ SC2230: which is non-standard. Use builtin 'command -v' instead.
    
    For more information:
      https://www.shellcheck.net/wiki/SC2230 -- which is non-standard. Use builti...
    
    Check shell style with shfmt................................................Passed
    CRLF end-lines remover......................................................Passed
    Check .sh files against bash-it requirements................................Passed
    Check .bash files against bash-it requirements..............................Passed
    Check that clean_files.txt is sorted alphabetically.........................Passed

**Comment by gaelicWizard** [comment]
I've never once used `command -v`; I always used `type -t`...but I don't know the difference! Which one is better?

File: completion/available/consul.completion.bash [hunk 1]
```diff
@@ -3,5 +3,5 @@ cite "about-completion"
 about-completion "Hashicorp consul completion"
 
 if _command_exists consul; then
-	complete -C "$(which consul)" consul
+	complete -C "$(command -v consul)" consul
```
Comment (NoahGorny) [inline completion/available/consul.completion.bash:6]: actually we can do `complete -C consul consul` here

File: themes/atomic/atomic.theme.bash [hunk 2]
```diff
@@ -148,7 +148,7 @@ ___atomic_prompt_ruby() {
 
 ___atomic_prompt_todo() {
 	[ "${THEME_SHOW_TODO}" != "true" ] \
-		|| [ -z "$(which todo.sh)" ] && return
+		|| [ -z "$(command -v todo.sh)" ] && return
```
Comment (NoahGorny) [inline themes/atomic/atomic.theme.bash:151]: we can do `|| ! _command_exists todo.sh` instead

File: themes/base.theme.bash [hunk 3]
```diff
@@ -85,10 +85,10 @@ RBENV_THEME_PROMPT_SUFFIX='|'
 RBFU_THEME_PROMPT_PREFIX=' |'
 RBFU_THEME_PROMPT_SUFFIX='|'
 
-GIT_EXE=$(which git 2> /dev/null || true)
-P4_EXE=$(which p4 2> /dev/null || true)
-HG_EXE=$(which hg 2> /dev/null || true)
-SVN_EXE=$(which svn 2> /dev/null || true)
+GIT_EXE=$(command -v git 2> /dev/null || true)
```
Comment (NoahGorny) [inline themes/base.theme.bash:88]: use `_command_exists`

File: themes/brainy/brainy.theme.bash [hunk 4]
```diff
@@ -136,7 +136,7 @@ ___brainy_prompt_ruby() {
 
 ___brainy_prompt_todo() {
 	[ "${THEME_SHOW_TODO}" != "true" ] \
-		|| [ -z "$(which todo.sh)" ] && return
+		|| [ -z "$(command -v todo.sh)" ] && return
```
Comment (NoahGorny) [inline themes/brainy/brainy.theme.bash:139]: same as the `atomic` theme (actually there is so much code dup here, we should split the logic to a separate file, but for now its enough)

File: themes/base.theme.bash [hunk 5]
```diff
@@ -400,7 +400,7 @@ function node_version_prompt {
 }
 
 function rvm_version_prompt {
-	if which rvm &> /dev/null; then
+	if command -v rvm &> /dev/null; then
```
Comment (NoahGorny) [inline themes/base.theme.bash:403]: `if _command_exists rvm; then`

File: themes/base.theme.bash [hunk 6]
```diff
@@ -409,7 +409,7 @@ function rvm_version_prompt {
 }
 
 function rbenv_version_prompt {
-	if which rbenv &> /dev/null; then
+	if command -v rbenv &> /dev/null; then
```
Comment (NoahGorny) [inline themes/base.theme.bash:412]: `_command_exists`

**Review by NoahGorny — CHANGES_REQUESTED** [review]
Hi @matchuek
I left a couple of comments. It seems like you are using an outdated `shellcheck` and this is why it generated all of those errors. It still great that you opened a PR, because there is still interesting cleanup things to do here!

Don't feel intimidated by the review! If you have any questions, feel free to comment and I will gladly help!

Good luck!

**Comment by NoahGorny** [comment]
> I've never once used `command -v`; I always used `type -t`...but I don't know the difference! Which one is better?

I actually prefer `type`, it seems to be safer all around. We actually have an helper function for this in Bash-it, called `_function_exists`/`_binary_exists` etc that use `type`

**Review by matchuek — COMMENTED** [review]
That's OK for me!

**Comment by NoahGorny** [comment]
> That's OK for me!

All right, whenever you have the time feel free to update the PR and I will take another look :smile: 

Thank you for contributing :heart:

**Comment by matchuek** [comment]
Hi @NoahGorny ,

> update the PR

what do you mean with that? I can't acknowledge your change request, github says:

_Pull request authors can’t approve their own pull request._

**Comment by NoahGorny** [comment]
> Hi @NoahGorny ,
> 
> > update the PR
> 
> what do you mean with that? I can't acknowledge your change request, github says:
> 
> _Pull request authors can’t approve their own pull request._

Hi @matchuek, I did not add the changes to your PR, I only suggested them. You need to commit them yourself, and push to this PR- than I can approve and merge :smile:

**Comment by matchuek** [comment]
I restarted with shellcheck 0.7.2 - everything passed so far!
I will retry on some of the many other files actually not in `/clean_files.txt`

**Pull request closed without merging**